cub/cub/device/device_run_length_encode.cuh

```diff
@@ -224,6 +224,12 @@ struct DeviceRunLengthEncode
 
     using policy_t = detail::device_run_length_encode_policy_hub<accum_t, key_t>;
 
+    if (num_items < 1)
+    {
+      *d_num_runs_out = 0;
+      return cudaSuccess;
+    }
```

Comment:
important: I'm afraid we cannot expect this to work reliably, as we may be writing to device memory from the host here. I think we'll need to assign `0` from within a kernel. 

Also, I think users expect to see no side-effects when `d_temp_storage` is `null`. So, I think we'd also need to check whether the user is just querying temporary storage requirements (i.e., `d_temp_storage` is `null`) or actually intends to invoke the algorithm.

Feel free to defer the actual fix and just file an issue to be picked up later on.

Reply:
Thanks for discovering this corner case! 🙏

Reply:
I backed this out and opened https://github.com/NVIDIA/cccl/issues/426